ReactNativeClient/lib/joplin-renderer/MdToHtml/rules/video.js

```diff
@@ -0,0 +1,31 @@
+module.exports = function videoPlugin(md) {
+
+	var defaultRender = md.renderer.rules.link_open;
+	const videoIDRegex = /^.*(youtu\.be\/|v\/|u\/\w\/|embed\/|watch\?v=|&v=)([^#&?]*).*/;
+	const key = /.*(youtube|youtu.be).*/im;
+
+	md.renderer.rules.link_open = function(tokens, idx, options, env, self) {
+		var token = tokens[idx];
+
+		var link = token.attrGet('href');
+
+		var match = link.match(key);
+		try {
+			if (match[1]==='youtube') {
+				let videoID = link.match(videoIDRegex);
+				let embedStart = '<div><iframe  width="560" height="315" src="https://www.youtube.com/embed/';
+				let embedEnd ='"  frameborder="0" allowfullscreen allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"></iframe></div>';
+
+				return embedStart + videoID[2] + embedEnd;
+
+			}
+		} catch (e) { return ''; }
+
+		// try-catch to remove uncaught error promises when match[1] is undefined
```

Comment:
Please don't silence exceptions, it leads to poor quality software since errors get undetected and never get fixed.

Secondly, the way I understand this code is like this: if the user adds a YouTube URL, display a video player. If they add any other URL, display nothing. Or am I missing something? You didn't add automated-tests yet but did you test with regular URLs in the app?

Reply:
yes, it checks if the domain name is youtube and then creates an iframe for the link, the `VideoIDRegex` extracts the 11 digit code which identifies a the youtube video ID and then concatenate it.

Reply:
> Secondly, the way I understand this code is like this: if the user adds a YouTube URL, display a video player. If they add any other URL, display nothing. Or am I missing something? You didn't add automated-tests yet but did you test with regular URLs in the app?

Reply:
Yes there is an inconsistency with regular links, 

With plugin disabled

<img width="1128" alt="Annotation 2020-03-22 193102" src="https://user-images.githubusercontent.com/54576074/77251330-c6c41700-6c73-11ea-8d19-8ac3dd8f3248.png">


Plugin enabled

<img width="1128" alt="link" src="https://user-images.githubusercontent.com/54576074/77251332-c9267100-6c73-11ea-8eac-a9011bd97468.png">


what do you think is causing this?